File: C# I/05. Conditional-Statements/ConditionalStatementsHomework/04. MultiplicationSign/MultiplicationSign.cs
using System;

class MultiplicationSign
{
    /*4. Multiplication Sign

    Write a program that shows the sign (+, - or 0) of the product
    of three real numbers, without calculating it.
    Use a sequence of if operators.
    */
    
    static void Main()
    {
        Console.Write("Enter (signed) numbre a: ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("Enter (signed) number b: ");
        double b = double.Parse(Console.ReadLine());
        Console.Write("Enter (signed) number c: ");
        double c = double.Parse(Console.ReadLine());
        bool foundZero = false;
        int minusSymbols = 0;
        if(a <= 0)
        {
            if(a == 0)
                foundZero = true;
            else
                minusSymbols++;
        }
        if(b <= 0)
        {
            if (b == 0)
                foundZero = true;
            else
                minusSymbols++;
        }
        if (c <= 0)
        {
            if (c == 0)
                foundZero = true;
            else
                minusSymbols++;
        }
        if(foundZero)
        {
            Console.WriteLine("Result: 0");
        }
        else
        {
            if(minusSymbols % 2 == 0)
            {
                Console.WriteLine("Result: +");
            }
            else
            {
                Console.WriteLine("Result: -");
            }
        }
    }
}

File: JavaScript/05. Conditional-Statements/ConditionalStatements-Homework/ConditionalStatements-Homework/02. MultiplicationSign/MultiplicationSign.js
﻿/*
 * 2. Multiplication Sign

Write a script that shows the sign (+, - or 0) of the product of three real numbers, without calculating it.
Use a sequence of if operators.
 */

var a = [5, -2, -2, 0, -1];
var b = [2, -2, 4, -2.5, -0.5];
var c = [2, 1, 3, 4, 5.1];

var result = "";

for (var i = 0; i < a.length; i += 1) {
    result = "+";
    if (a[i] < 0 || b[i] < 0 || c[i] < 0) {
        result = "-";
    }
    if ((a[i] < 0 && b[i] < 0) || (a[i] < 0 && c[i] < 0) || (b[i] < 0 && c[i] < 0)) {
        result = "+";
    }
    if (a[i] < 0 && b[i] < 0 && c[i] < 0) {
        result = "-";
    }
    if (a[i] == 0 || b[i] == 0 || c[i] == 0) {
        result = "0";
    }
    console.log("a: " + a[i] + " b: " + b[i] + " c: " + c[i] + " result: " + result);
}